services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/EventController.java

```diff
@@ -0,0 +1,1023 @@
+            list.add(MockEventGenerator.getSpatBroadcastRateEvent());
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = spatBroadcastRateEventRepo.getQuery(intersectionID, startTime, endTime, latest);
+            return ResponseEntity.ok(spatBroadcastRateEventRepo.find(query));
+        }
+    }
+
+    @RequestMapping(value = "/events/spat_broadcast_rate/count", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("@PermissionService.isSuperUser() || (@PermissionService.hasIntersection(#intersectionID) and @PermissionService.hasRole('USER')) ")
+    public ResponseEntity<Long> countSpatBroadcastRateEvents(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "full_count", required = false, defaultValue = "true") boolean fullCount,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            return ResponseEntity.ok(1L);
+        } else {
+            Query query = spatBroadcastRateEventRepo.getQuery(intersectionID, startTime, endTime, false);
+
+            long count;
+
+            if (fullCount) {
+                count = spatBroadcastRateEventRepo.getQueryFullCount(query);
+            } else {
+                count = spatBroadcastRateEventRepo.getQueryResultCount(query);
+            }
+
+            log.debug("Found: {} SpatBroadcastRateEvents", count);
+            return ResponseEntity.ok(count);
+        }
+    }
+
+    @RequestMapping(value = "/events/spat_message_count_progression", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("hasRole('USER') || hasRole('ADMIN')")
+    public ResponseEntity<List<SpatMessageCountProgressionEvent>> findSpatMessageCountProgressionEvents(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "start_time_utc_millis", required = false) Long startTime,
+            @RequestParam(name = "end_time_utc_millis", required = false) Long endTime,
+            @RequestParam(name = "latest", required = false, defaultValue = "false") boolean latest,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+
+        if (testData) {
+            List<SpatMessageCountProgressionEvent> list = new ArrayList<>();
+            list.add(MockEventGenerator.getSpatMessageCountProgressionEvent());
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = spatMessageCountProgressionEventRepo.getQuery(intersectionID, startTime, endTime, latest);
+            return ResponseEntity.ok(spatMessageCountProgressionEventRepo.find(query));
```

Comment:
issue(blocking): executing query second time for logging purposes

Reply:
Extra query has been removed